open-apparel-registry/open-apparel-registry
# Add event model

## Overview

Adds the Event model and corresponding migration. This new model will
be the foundation of the change notification system and a revised
history API. It makes use of contenttypes to allow us to use a generic
foreign field and to delete the associated objects without losing
the Event object or the ID of the deleted object.

Connects #1989 

## Demo

Optional. Screenshots, `curl` examples, etc.

## Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* `./scripts/manage migrate`
* `./scripts/manage shell_plus`
* In the shell:
```
from datetime import datetime
facility = Facility.objects.first()
event = Event.objects.create(content_object=facility, event_type='OTHER', event_details='testing', event_time=datetime.now())
event
event.content_object
event_2 = Event.objects.create(content_object=FacilityListItem.objects.first(), event_type='OTHER', event_details='testing', event_time=datetime.now())
event_2
event_2.content_object
```
* You should see events printed to the console, and the related Facility and FacilityListItem objects printed as well. Note the ID of the Facility.
* Close the shell. Run `./scripts/server` and login to the site as c1@example.com. Use the [Dashboard Facility Delete](http://localhost:6543/dashboard/deletefacility) to delete the facility.
* Restart `shell_plus`. In the shell:
```
event = Event.objects.get(id=1)
event
event.content_object
event.object_id
```
* You should see the event printed and the ID of the facility printed, but no Facility object should be printed. There should be no error. 

## Checklist
